#! /bin/sh /usr/share/dpatch/dpatch-run
## 02_more_verbose_toram.dpatch by  <mika@grml.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Support more verbose version of toram via rsync

@DPATCH@

--- a/scripts/live	Wed Feb 20 16:00:41 2008 +0100
+++ b/scripts/live	Wed Feb 20 17:57:04 2008 +0100
@@ -593,8 +593,7 @@ copy_live_to ()
 
 	if [ "${freespace}" -lt "${size}" ]
 	then
-		[ "${quiet}" != "y" ] && log_warning_msg "Not enough free ${free_string} (${freespace}k free, ${size}k needed) to copy live media in ${copytodev}."
-		return 1
+		echo " * Warning: Not enough free ${free_string} (${freespace}k free, ${size}k needed) to copy live media in ${copytodev}." >/dev/console
 	fi
 
 	# begin copying (or uncompressing)
@@ -612,9 +611,19 @@ copy_live_to ()
 	else
 		if [ -n "${MODULETORAMFILE}" ]
 		then
-			cp ${MODULETORAMFILE} ${copyto} # copy only the filesystem module
-		else
-			cp -a ${copyfrom}/* ${copyto}   # "cp -a" from busybox also copies hidden files
+			if [ -x /usr/bin/rsync ] ; then
+			echo " * Copying $MODULETORAMFILE to RAM" 1>/dev/console
+			   rsync -a --progress ${MODULETORAMFILE} ${copyto} 1>/dev/console # copy only the filesystem module
+			else
+			   cp ${MODULETORAMFILE} ${copyto} # copy only the filesystem module
+			fi
+        	else
+			if [ -x /usr/bin/rsync ] ; then
+			echo " * Copying medium to RAM" 1>/dev/console
+			   rsync -a --progress ${copyfrom}/* ${copyto} 1>/dev/console  # "cp -a" from busybox also copies hidden files
+			else
+			   cp -a ${copyfrom}/* ${copyto}   # "cp -a" from busybox also copies hidden files
+			fi
 		fi
 
 		livefs_root
